### PR TITLE
Parser Behaviour & Yarn Support (Issue #80)

### PR DESCRIPTION
**Pull request by haldihri3**

### CODE REVIEW

File: lib/parser.ex [hunk 1]
```diff
@@ -0,0 +1,4 @@
+defmodule Parser do
```
Comment (kitplummer) [inline lib/parser.ex:1]: This, and any new files need the copyright header.

File: test/npm/yarnlock_test.exs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+defmodule YarnlockTest do
+  use ExUnit.Case
+
+  test "extracts dependencies from yarn.lock" do
+    {lib_map, deps_count} = Npm.Yarnlockfile.parse!(File.read!("./test/fixtures/yarnlock"))
+
+    parsed_yarn = [{"ajv", "6.12.3"}, {"asn1", "0.2.4"}, {"assert-plus", "1.0.0"}]
```
Comment (kitplummer) [inline test/npm/yarnlock_test.exs:7]: I didn't nit pick in the other test cases, but can we add an assertion to verify the actual library and version for one of them?

File: test/project_ident_test.exs [hunk 3]
```diff
@@ -11,7 +11,7 @@ defmodule ProjectIdentTest do
 
     mix_type = %ProjectType{name: :mix, path: "**", files: ["mix.exs,mix.lock"]}
     python_type = %ProjectType{name: :python, path: "**", files: ["setup.py,*requirements.txt*"]}
-    node_type = %ProjectType{name: :node, path: "**", files: ["package*.json"]}
+    node_type = %ProjectType{name: :node, path: "**", files: ["package*.json,yarn.lock"]}
```
Comment (kitplummer) [inline test/project_ident_test.exs:14]: Is there a case where a project might have both?  Can't imagine, but also not positive.

File: lib/npm/yarnlockfile.ex [hunk 4]
```diff
@@ -0,0 +1,40 @@
+defmodule Npm.Yarnlockfile do
```
Comment (kitplummer) [inline lib/npm/yarnlockfile.ex:1]: Need the copyright header

File: lib/npm/packagefile.ex [hunk 5]
```diff
@@ -1,24 +1,31 @@
 defmodule Npm.Packagefile do
```
Comment (kitplummer) [inline lib/npm/packagefile.ex:1]: Need copyright header

**Comment by haldihri3** [comment]
> Is there a case where a project might have both?  Can't imagine, but also not positive. 

A project can have both a package-lock.json and a yarn.lock to provide support for both NPM and Yarn.  Currently, if both exist, we scan the package-lock.json only

File: test/npm/yarnlock_test.exs (hunk 2 above)
Comment (kitplummer) [reply test/npm/yarnlock_test.exs:7]: Durr...i see it now.  Sheesh, where's my glasses?

**Pull request merged**